ucsb-cs156-s22/s22-6pm-happycows
# s22-6pm-3 - dj -  add countdown to commons card

Replaced by https://github.com/ucsb-cs156/proj-happycows/pull/1

# Overview

I've added a countdown in the CommonsCard on the Login and Home pages.

# Issues Addressed

This closes Issue #78 (add the endDate countdown for each Commons to the login page)

# Story

As a user (logged in or out), I can see how many days are left before each Commons is finished.

# Details

The Commons tables on the Login and Home pages now have a new column, called "Days Left", under which the amount of days before the end date for each commons is listed. After the end date for a commons, it says "Finished" instead. If the end date was never specified, the table says "Not set" for the amount of days left.

![image](https://user-images.githubusercontent.com/72847482/171633259-83142115-8bfa-4b93-86a9-894c2649ad65.png)
![image](https://user-images.githubusercontent.com/72847482/171633315-cdf207ae-6f06-4b19-92ac-8e847350f036.png)


# Test Plan (for interactive testing)

* Open happycows in localhost:8080, and log in as an admin
* Use Swagger UI to add new Commons, with end dates before and after the current date. Feel free to also ignore the end date field. The ones ending before the current date should say "Finished", while the ones ending afterwards should list the number of days before then.
* Use the Create Commons page to make a new commons. Currently, that page doesn't have a field for the end date. After those commons are created, their cards in the home page should say "Not set" for the "Days Left" column.
* Log out. The table with the data should still be there to see, but without the join button.

Codecov:
![image](https://user-images.githubusercontent.com/72847482/171632263-08de55dc-52b7-4dfb-977c-5f15794f7500.png)
![image](https://user-images.githubusercontent.com/72847482/171632327-485e035e-2b8a-4140-b59d-aa2fa7d3818a.png)

Stryker:
![image](https://user-images.githubusercontent.com/72847482/171631752-7c88171d-db95-453c-8222-97d4a47ddc08.png)
